sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -529,69 +603,178 @@ myparams::joint_matrix_b<sub_group, layout::row_major> sub_b;
+enum class matrix_type {
+  bf16,
+  fp16,
+  tf32,
+  fp32,
+  fp64,
+  sint8,
+  sint16,
+  sint32,
+  sint64,
+  uint8,
+  uint16,
+  uint32,
+  uint64
+};
+struct combination {
+  uint32_t max_msize;
+  uint32_t max_nsize;
+  uint32_t max_ksize;
+  uint32_t msize;
+  uint32_t nsize;
+  uint32_t ksize;
+  matrix_type atype;
+  matrix_type btype;
+  matrix_type accumulatortype;
+};
 
-```c++
-auto data = get_wi_data(sg, C);
-for (int i = 0; i < data.length(); ++i) {
-  sum_of_local_rows[row] += data[i];
-}
+} // namespace sycl::ext::oneapi::experimental::matrix
 ```
 
-We want to keep backward compatibility in the joint matrix code when implementations or hardware change. To that end, instead of hard-coding this mapping, we use general backend and target-agnostic functionality, especially in the JIT compilation mode of SYCL. For this reason we would like to be able to query this mapping so that code does not have to change from one version to the other.
+Each combination of the `combinations` vector composes the types and
+sizes of A, B, accumulator matrices supported by the device
+implementation. The
+table below provides a description of each member of the `combination` struct.
 
-So for the mapping problem, since this mapping is implementation-defined, one of the proposals is to add runtime functions like:
+[frame="none",options="header"]
+|======================
+| Member of `combination` | Description
+|`max_msize`, `max_nsize`, `max_ksize`| if the matrix implementation
+supports a continuous number of element sizes, each of these members
+is non-zero, and the matrix implementation supports all element sizes
+from 1 up to (and including) that number. By contrast, if the matrix
+hardware implementation supports a discrete number of element sizes,
+each of these members has the value zero
+|`msize`, `nsize`, `ksize`| if the matrix implementation supports a
+discrete number of element sizes, each of these members is non-zero,
+and the value tells one of the supported element sizes. By contrast,
+if the matrix hardware supports a continuous number of element sizes,
+each of these members has the value zero
+|`atype`, `btype`, `accumulatortype`| indicates the types supported in
+the combination. these are of type `matrix_type` which tells the list
+of types that are supported for the A, B, and accumulator matrices in
+the `T` template parameter as follows: +
+`bf16`: `sycl::bfloat16` +
+`fp16`: `sycl::half` +
+`tf32`: `sycl::ext::oneapi::experimental::matrix::precision::tf32` +
+`fp32`: `float` +
+`fp64`: `double` +
+`sint8`: signed 8 bits signed integer +
+`sint16`: `signed short` +
+`sint32`: `signed int` +
+`sint64`: `signed long` +
+`uint8`: unsigned 8 bits integer +
+`uint16`: `unsigned short` +
+`uint32`: `unsigned int` +
+`uint64`: `unsigned long` 
```

Comment:
Strictly speaking, `short` is not necessarily a 16-bit type, `int` is not necessarily a 32-bit type, and `long` is not necessarily a 64-bit type.  It's probably better to show the fixed-width types in this list instead.  For example `sint8` corresponds to `int8_t`, and `uint64` corresponds to `uint64_t`.